lib/puppet/util/logging.rb

```diff
@@ -194,9 +194,9 @@ def get_deprecation_offender()
     # let's find the offending line;  we need to jump back up the stack a few steps to find the method that called
     #  the deprecated method
     if Puppet[:trace]
-      caller()[2..-1]
+      caller(4)
```

Comment:
Someone should double check my math on this and the one below.

Reply:
The default signature is `Kernel.caller(start=1, length=nil)`, so previously we skipped the first frame (index 0), and then skipped array entries 0 and 1. So I think we want `caller(3)` which should skip the first 3 frames (0,1,2)?

Reply:
Thank you, when I rebased I noticed the previous version was incorrect, but everytime I did it in my head I got a different answer.